src/other/tables.jl

```diff
@@ -48,6 +48,15 @@ DataFrame!(x::Vector{<:NamedTuple}) =
                         "`$(typeof(x))` without allocating new columns: use " *
                         "`DataFrame(x)` instead"))
 
+const _DataFrameIter = Union{DataFrameRows, DataFrameColumns}
```

Comment:
Intuitively I would prefer to use `@eval` here on two types, as then we do not have to introduce an unexported union that users might not understand when using `methods` function.